src/libexpr/primops.cc

```diff
@@ -878,6 +878,33 @@ static RegisterPrimOp primop_trace({
 });
 
 
+/* Evaluate the first expression and print it on standard error if talkative verbosity
+ * is enabled. Then return the second expression.  Useful for debugging. */
+static void prim_debug(EvalState & state, const Pos & pos, Value * * args, Value & v)
+{
+    state.forceValue(*args[0], pos);
```

Comment:
Notably this causes Nix evaluation to slightly differ if verbose mode is enabled, but I think it's not a problem because the result here doesn't affect the evaluation result, other than if the expression fails to evaluate. I think this is a worthwhile tradeoff for having an essentially no-cost `builtins.traceVerbose` we can spam in the codebase for debugging purposes.